packages/dds/tree/src/feature-libraries/flex-map-tree/mapTreeNode.ts

```diff
@@ -407,36 +416,65 @@ class EagerMapTreeField<T extends FlexAllowedTypes> implements MapTreeField {
 	}
 
 	public context: undefined;
-}
 
-class EagerMapTreeRequiredField<T extends FlexAllowedTypes>
-	extends EagerMapTreeField<T>
-	implements FlexTreeRequiredField<T>
-{
-	public get editor(): ValueFieldEditBuilder<ExclusiveMapTree> {
-		throw unsupportedUsageError("Setting a required field");
-	}
-
-	public get content(): FlexTreeUnboxNodeUnion<T> {
-		return unboxedUnion(this.schema, this.mapTrees[0] ?? oob(), { parent: this, index: 0 });
+	// All edits to the field (i.e. mutations of the field's MapTrees) should be directed through this function.
+	// This function ensures that the parent MapTree has no empty fields (which is an invariant of `MapTree`) after the mutation.
+	protected edit(mutation: (mapTrees: ExclusiveMapTree[]) => void): void {
+		const mapTrees = this.parent.mapTree.fields.get(this.key) ?? [];
+		mutation(mapTrees);
+		if (mapTrees.length > 0) {
+			this.parent.mapTree.fields.set(this.key, mapTrees);
+		} else {
+			this.parent.mapTree.fields.delete(this.key);
+		}
 	}
 }
 
 class EagerMapTreeOptionalField<T extends FlexAllowedTypes>
 	extends EagerMapTreeField<T>
 	implements FlexTreeOptionalField<T>
 {
-	public get editor(): OptionalFieldEditBuilder<ExclusiveMapTree> {
-		throw unsupportedUsageError("Setting an optional field");
-	}
+	public editor = {
+		set: (newContent: ExclusiveMapTree | undefined) => {
+			// If the new content is a MapTreeNode, it needs to have its parent pointer updated
+			if (newContent !== undefined) {
+				nodeCache.get(newContent)?.adoptBy(this, 0);
```

Comment:
Yes - anytime the MapTree originated from POJO that was not part of a node's constructor, you will get undefined here. E.g. if you did:

```
foo.bar = [a, b, c]; // `[a, b, c]` will become a MapTree but does not have a corresponding unhydrated node
map.set("key", { baz: 3 }); // `{ baz: 3 }` will become a MapTree but does not have a corresponding unhydrated node
```

Reply:
For existing content, it will always have a MapTree node _in practice_, but that's only because we currently do a full demand walk of unhydrated nodes when they are created. It's not a necessary invariant though.